resolwe/flow/managers/listener/listener.py

```diff
@@ -69,22 +69,155 @@
 logger = logging.getLogger(__name__)
 
 
-class Processor:
-    """Class respresents instance of one connected worker."""
+class RedisCache:
+    """The Redis cache."""
+
+    cached_fields_type = {
```

Comment:
It would be nice to cache the outputs of finished processes.

Reply:
I agree. When they are used in a pipeline, the output of already finished process is very likely to be used in the near future. But this would require changes in the RedisCache since output can be deeply nested and use various types for values.

How about using lru_cache to store the last 1000 outputs for example? The cache will be local to every listener instance, but there will probably not be a lot of them and the output of already finished processes can not change.